app/src/main/java/org/dhis2/utils/RulesUtilsProviderImpl.kt

```diff
@@ -145,19 +137,21 @@ class RulesUtilsProviderImpl(private val codeGenerator: CodeGenerator) : RulesUt
     ) {
         val model = fieldViewModels[showWarning.field()]
         if (model != null) {
-            fieldViewModels[showWarning.field()] = model.withWarning(showWarning.content() + data)
+            fieldViewModels[showWarning.field()] =
+                model.withWarning(showWarning.content() + " " + data)
         }
     }
 
     private fun showError(
         showError: RuleActionShowError,
         fieldViewModels: MutableMap<String, FieldViewModel>,
-        rulesActionCallbacks: RulesActionCallbacks
+        rulesActionCallbacks: RulesActionCallbacks,
+        data: String
     ) {
         val model = fieldViewModels[showError.field()]
 
         if (model != null) {
-            fieldViewModels[showError.field()] = model.withError(showError.content())
+            fieldViewModels[showError.field()] = model.withError(showError.content() + " " + data)
```

Comment:
With kotlin we can use String interpolation to concatenate strings

`fieldViewModels[showError.field()] = model.withError("${showError.content()} $data")`

when you have to use a method or property of the variable it needs to be wrapped with the { } signs.

Reply:
Done